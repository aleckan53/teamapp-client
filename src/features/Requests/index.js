import React, { useContext } from 'react'
import RequestCard from './components/RequestCard'
import { Header, Msg } from '../Basic'
import context from 'context'

const Requests = props => {

  const { incoming, outgoing } = useContext(context.Events).requests

  return (
    <section className='Requests'>
      <Header h1='Requests'/>
      { !incoming.length ? '' : (
        <div>
          <h3>Incoming</h3>
          {incoming.map((r,i)=> 
            <RequestCard.Incoming {...r} key={i}/>
          )}
        </div>
      )}  
      { !outgoing.length ? '' : (
        <div>
          <h3>Outgoing</h3>
          {outgoing.map((r,i)=> 
            <RequestCard.Outgoing {...r} key={i}/>
          )}
        </div>
      )}
      { !incoming.length && !outgoing.length ? <Msg text="New requests appear here"/> : '' }
    </section>
  )
}

export default Requests