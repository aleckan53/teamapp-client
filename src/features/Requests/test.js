import React from 'react'
import ReactDOM from 'react-dom'
import Requests from './index.js'
import { create } from 'react-test-renderer'

describe('Account component', () => {
  it('renders without crushing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Requests/>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  describe('Given no requests', () => {
    it('renders a message', () => {
      const component = create(<Requests/>)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
