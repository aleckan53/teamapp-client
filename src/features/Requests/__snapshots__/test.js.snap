// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component Given no requests renders a message 1`] = `
<section
  className="Requests"
>
  <header
    className="header"
  >
    <h1
      className="h1"
    >
      Requests
    </h1>
  </header>
  
  
  <p
    className="msg"
  >
    New requests appear here
  </p>
</section>
`;
